libs/core/include/core/bitvector.hpp

```diff
@@ -48,6 +48,8 @@ class BitVector
   BitVector(BitVector const &other);
   ~BitVector() = default;
 
+  BitVector &operator=(BitVector const &other);
```

Comment:
Worth adding a `BitVector &operator=(BitVector &&other) noexcept = default` to be clear (or maybe I have missed the point?

Also I would probably do a move constructor too

Reply:
(Explicitly defaulted are implicitly noexcept.)